superset-frontend/src/SqlLab/components/SqlEditor/index.tsx

```diff
@@ -0,0 +1,742 @@
+import { useDispatch, useSelector } from 'react-redux';
+import Split from 'react-split';
+import { t, styled, useTheme } from '@superset-ui/core';
+import debounce from 'lodash/debounce';
+import throttle from 'lodash/throttle';
+import StyledModal from 'src/components/Modal';
+import Mousetrap from 'mousetrap';
+import Button from 'src/components/Button';
+import Timer from 'src/components/Timer';
+import {
+  Dropdown,
+  Menu as AntdMenu,
+  Menu,
+  Switch,
+  Input,
+} from 'src/common/components';
+import { DatabaseObject } from 'src/components/DatabaseSelector';
+import { QueryEditor, Query } from 'src/SqlLab/types';
+import Icons from 'src/components/Icons';
+import { detectOS } from 'src/utils/common';
+import {
+  addQueryEditor,
+  CtasEnum,
+  estimateQueryCost,
+  persistEditorHeight,
+  postStopQuery,
+  queryEditorSetAutorun,
+  queryEditorSetQueryLimit,
+  queryEditorSetSql,
+  queryEditorSetTemplateParams,
+  runQuery,
+  saveQuery,
+  addSavedQueryToTabState,
+  scheduleQuery,
+  setActiveSouthPaneTab,
+  updateSavedQuery,
+  validateQuery,
+} from 'src/SqlLab/actions/sqlLab';
+import {
+  STATE_TYPE_MAP,
+  SQL_EDITOR_GUTTER_HEIGHT,
+  SQL_EDITOR_GUTTER_MARGIN,
+  SQL_TOOLBAR_HEIGHT,
+} from 'src/SqlLab/constants';
+import {
+  getItem,
+  LocalStorageKeys,
+  setItem,
+} from 'src/utils/localStorageHelpers';
+import { FeatureFlag, isFeatureEnabled } from 'src/featureFlags';
+import rootReducer from 'src/SqlLab/reducers';
+import TemplateParamsEditor from '../TemplateParamsEditor';
+import ConnectedSouthPane from '../SouthPane/state';
+import SaveQuery from '../SaveQuery';
+import ScheduleQueryButton from '../ScheduleQueryButton';
+import EstimateQueryCostButton from '../EstimateQueryCostButton';
+import ShareSqlLabQuery from '../ShareSqlLabQuery';
+import SqlEditorLeftBar from '../SqlEditorLeftBar';
+import AceEditorWrapper from '../AceEditorWrapper';
+import RunQueryActionButton from '../RunQueryActionButton';
+
+type RootState = ReturnType<typeof rootReducer>;
+
+const LIMIT_DROPDOWN = [10, 100, 1000, 10000, 100000];
+const SQL_EDITOR_PADDING = 10;
+const INITIAL_NORTH_PERCENT = 30;
+const INITIAL_SOUTH_PERCENT = 70;
+const SET_QUERY_EDITOR_SQL_DEBOUNCE_MS = 2000;
+const VALIDATION_DEBOUNCE_MS = 600;
+const WINDOW_RESIZE_THROTTLE_MS = 100;
+
+const appContainer = document.getElementById('app');
+const bootstrapData = JSON.parse(
+  appContainer.getAttribute('data-bootstrap') || '{}',
+);
+const validatorMap =
+  bootstrapData?.common?.conf?.SQL_VALIDATORS_BY_ENGINE || {};
+const scheduledQueriesConf = bootstrapData?.common?.conf?.SCHEDULED_QUERIES;
+
+interface SqlEditorProps {
+  actions: object;
+  database?: DatabaseObject;
+  latestQuery?: Query;
+  tables: any[];
+  editorQueries: QueryEditor[];
+  dataPreviewQueries: object[];
+  queryEditorId: string;
+  hideLeftBar?: boolean;
+  defaultQueryLimit: number;
+  maxRow: number;
+  displayLimit: number;
+  saveQueryWarning?: string;
+  scheduleQueryWarning?: string;
```

Comment:
Do you think that for `actions` `tables` `dataPreviewQueries` we can be a little more specific for their types?

Reply:
It's better do in next PR

Reply:
@geido Do we have types for all sqlLab `actions` & `dataPreviewQueries`?

Reply:
@EugeneTorap if you can't find any reference in the code, we probably don't have but I am happy to do that in a related PR!